engine.py

```diff
@@ -114,7 +115,26 @@ def _on_schema_loaded():
         # Python processes that get spawned. By setting it at the top, it'll be
         # propagated down to any of Photoshop's subprocesses.
         if "TANK_CONTEXT" in os.environ:
-            self.adobe.dollar.setenv("TANK_CONTEXT", self.context.serialize())
+            self.adobe.dollar.setenv("TANK_CONTEXT", new_context.serialize())
+
+        # If there's an active document and it isn't already recorded in the
+        # context cache, go ahead and add it.
+        active_doc = self.adobe.get_active_document_path()
+
+        if active_doc and active_doc not in self._CONTEXT_CACHE:
```

Comment:
Would it be ok to always update the cache for the new context if there is an active document? Are there any scenarios where the existing context could be empty, incorrect, or incomplete? Maybe not, but just wondering if the check for existence in the cache is necessary.